### PR TITLE
Pdf style added

### PR DESCRIPTION
**Pull request by abkarim**

### CODE REVIEW

File: src/ContextMenu.tsx [hunk 1]
```diff
@@ -40,7 +41,7 @@ const ContextMenu: React.FC<ContextMenuProps> = ({ closeModal, content }) => {
   }, [contextMenuRef]);
 
   const handleDownloadPDF = useCallback(() => {
-    downloadPDF(content);
+    downloadPDF(element);
```
Comment (oikantik) [inline src/ContextMenu.tsx:48]: 🤔 should this be content?

File: src/Content.tsx [hunk 2]
```diff
@@ -53,17 +60,24 @@ const handleExportContextMenuClick = (event: MouseEvent) => {
       )
   );
 
+  // Ignore this element on pdf download
+  menuContainer?.setAttribute('data-html2canvas-ignore', 'true');
```
Comment (oikantik) [inline src/Content.tsx]: The `menuContainer` above this is ugly af (my fault 😅 ). 

You can add something like this in the helper file

```javascript
export const findNearestElementById = (element: Element | null | undefined, targetId: string): Element | null => {
  if (!element) {
    return null;
  }

  // Search for siblings with the target ID
  const sibling = element?.parentElement?.querySelector(`#${targetId}`);
  if (sibling) {
    return sibling;
  }

  // Continue searching up the DOM tree
  return findNearestElementById(element.parentElement, targetId);
}
```

and then use it like this

```javascript
const menuContainer = findNearestElementById(target, MENU_CONTAINER);
```

File: src/Content.styles.tsx [hunk 3]
```diff
@@ -85,15 +85,18 @@ export const EditorContainer = styled.div`
     border: none;
     border-bottom: ${border};
   }
-
+  
   & .ql-container.ql-snow {
     border: none;
+    color: black !important;
```
Comment (oikantik) [inline src/Content.styles.tsx]: Use hex please!

File: src/Content.tsx [hunk 4]
```diff
@@ -7,13 +7,15 @@ import {
...
   BUTTON_CONTAINER,
   CONTENT_CONTAINER,
-  DOWNLOAD_SVG,
   MAIN_CONTAINER,
   MODAL_CONTAINER,
   MENU_CONTAINER,
   CHAT_CONTAINER,
 } from "./utils/constants";
 
+import downloadImage from "./icons/download.png";
```
Comment (oikantik) [inline src/Content.tsx:16]: Can we use a better download image please? this one does not look so good

Screen Shot 2023-03-27 at 9 21 37 PM

File: src/Popup.tsx [hunk 5]
```diff
@@ -25,11 +24,18 @@ const Popout = () => {
         const activeTab = tabs[0];
 
         if (activeTab.id) {
-          chrome.tabs.sendMessage(
+          if(type === "PDF") {
+            chrome.tabs.sendMessage(
+              activeTab.id,
+              {message: 'download_full_page'}  
+            )
+          }else {
```
Comment (oikantik) [inline src/Popup.tsx]: Make sure to run prettier on changed files please

File: src/utils/helpers.ts [hunk 6]
```diff
@@ -17,30 +17,38 @@ export const generateDownload = (
   URL.revokeObjectURL(url);
 };
 
-// TODO: Make filename timestamp.filetype
-export const downloadPDF = (content: string) => {
+const filename = () => {
```
Comment (oikantik) [inline src/utils/helpers.ts:20]: We should probably add a timestamp, something like this

```javascript
  const now = new Date();
  const year = now.getFullYear();
  const month = String(now.getMonth() + 1).padStart(2, '0');
  const day = String(now.getDate()).padStart(2, '0');
  const hour = String(now.getHours()).padStart(2, '0');
  const minute = String(now.getMinutes()).padStart(2, '0');
  const second = String(now.getSeconds()).padStart(2, '0');

  const timestamp = `${year}${month}${day}${hour}${minute}${second}`;
  
  ```

File: src/utils/helpers.ts [hunk 7]
```diff
@@ -17,30 +17,38 @@ export const generateDownload = (
...
+  const title = document.title
+  return title
+}
+
+export const downloadPDF = (content: any) => {
   html2pdf(content, {
     margin: 0,
-    filename: "prompt.pdf",
+    filename: filename() + '.pdf',
```
Comment (oikantik) [inline src/utils/helpers.ts]: well while adding inline function invocation works, I'd prefer to assign it to a variable and then use it. Also lets keep using template literals so it's consistent

e.g.

```javascript
const name = filename()
...
filename: `${name}.pdf`
```

File: src/ContextMenu.tsx (hunk 1 above)
Comment (abkarim) [reply src/ContextMenu.tsx:48]: If we use content (innerHTML) PDF does not take page style and html2pdf requires a element to style

I don't know why But content doesn't take style 

Only for styling we moved to this solution

File: src/Content.styles.tsx (hunk 3 above)
Comment (abkarim) [reply src/Content.styles.tsx]: Sure

File: src/Content.tsx (hunk 2 above)
Comment (abkarim) [reply src/Content.tsx]: Thank you

File: src/Content.tsx (hunk 4 above)
Comment (abkarim) [reply src/Content.tsx:16]: Yeah sure

File: src/utils/helpers.ts (hunk 6 above)
Comment (abkarim) [reply src/utils/helpers.ts:20]: Sure

**Pull request merged**